ai_diffusion/ui/widget.py

```diff
@@ -599,30 +659,52 @@ def __init__(self, slider_range: tuple[int, int] = (1, 100), parent=None):
         self._layout.setContentsMargins(0, 0, 0, 0)
         self.setLayout(self._layout)
 
+        self._model = None
+
         self._slider = QSlider(Qt.Orientation.Horizontal, self)
         self._slider.setMinimum(slider_range[0])
         self._slider.setMaximum(slider_range[1])
-        self._slider.setSingleStep(5)
-        self._slider.valueChanged.connect(self.notify_changed)
+        self._slider.setSingleStep(1)
```

Comment:
Why this change? I like 5. It means scroll-wheeling over the slider changes in steps of 10, which feels nice. With singleStep=1 scroll-wheeling over the slider doesn't work anymore at all in a lot of situations (threshold interval > 5).

Technically this is still a problem when total steps is as low as 4, because in that case 10 won't be enough to jump to the next threshold, but it's a rare case and can be accepted...

Reply:
I wanted to avoid overscrolling (two steps at a time) when for instance the natural step size is 3. But since I don't scroll-wheel the slider myself anyways, I guess I don't mind. Done.

Reply:
Note that I still have the input box set to singleStep(1), cause there it'd really be weird to have "arrow-up" go two steps.

Reply:
Yea it's not an issue for the spin box since it supports custom step function. The way snap is done for the slider unfortunately means if you use arrow keys or scroll wheel and don't reach the next threshold, it always snaps back to the previous value, and never changes at all.

Anyway let's hope for now this solution is good enough and nobody actually uses arrow keys.

Reply:
In case I forgot to mention it: the reason I do it with `value` is that I tried overriding the mouse move event and that makes the UI so slow it becomes unusable. So I basically can't touch those methods with python which means I can't notice *why* the value changed. Maybe it has something like stepBy I can override?

Reply:
But yeah probably nobody's using it.

Reply:
Maybe it's possible by overriding [`QSlider.triggerAction`](https://doc.qt.io/qt-6/qabstractslider.html#triggerAction) but I haven't tried.

I will merge it later today if you don't plan further changes.

Reply:
Cheers! No, far as I can tell it's good.

Reply:
Done, already getting used to it :)
Thanks for all the work and I hope you didn't get burned too hard by all the Qt signal stuff!

Reply:
It was definitely a handful, but I think it ended up well for it.

I worry a bit that the code seemed reasonable on the first try, and then there turned out to be many changes necessary where at the end it still seems reasonable, and even a bunch simpler, but also it turns out the exact way it looks is exactly how it *had* to look and no other way was going to work on pain of terrible crashes. But honestly, I'm gonna stop thinking about it now. :)